File: packages/website/src/lib/types/enums.ts
export type StationType = 'A' | 'B';
export type AssetStatus = 'active' | 'repair';
export type DockState = 'empty' | 'occupied';
export type BikeType = 'electric' | 'conventional';
export type RunType = 'station' | 'detail';
export type ExitStatus = 'success' | 'partial' | 'error';
export type PeakPeriod = 'morning' | 'evening' | 'off_peak';
export type PeakStatus = 'DESERT' | 'OK';
export type DegradationStatus = 'STRUCTURALLY_DEGRADED' | 'DEGRADED_WATCH' | 'OK';
export type DeadDockStatus = 'FLAGGED' | 'OK';
export type BikeActivityStatus = 'ACTIVE' | 'STATIONARY' | 'IDLE_SUSPICIOUS';

/** Semantic health thresholds for KPI cards */
export type HealthLevel = 'critical' | 'warning' | 'good' | 'excellent';

/** Supported locales */
export type Locale = 'pt' | 'en' | 'es' | 'fr';

/** Time range presets available in filters */
export type TimeRangePreset = '24h' | '7d' | '30d' | '90d' | 'all';

/** Day type filter */
export type DayType = 'weekday' | 'weekend' | 'all';

File: packages/scraper/src/scraper/domain/enums.py
"""Domain enums matching DuckDB CREATE TYPE definitions in 001_initial_schema.sql."""

import enum


class StationStatus(enum.StrEnum):
    """Station operational status. Matches SQL: station_status."""

    ACTIVE = "active"
    REPAIR = "repair"


class StationType(enum.StrEnum):
    """Station hardware type. Matches SQL: station_type."""

    A = "A"
    B = "B"


class DockState(enum.StrEnum):
    """Dock occupancy state. Matches SQL: dock_state."""

    EMPTY = "empty"
    OCCUPIED = "occupied"


class BikeType(enum.StrEnum):
    """Bike propulsion type. Matches SQL: bike_type."""

    ELECTRIC = "electric"
    CONVENTIONAL = "conventional"


class RunType(enum.StrEnum):
    """Scrape run type. Matches SQL: run_type."""

    STATION = "station"
    DETAIL = "detail"


class ExitStatus(enum.StrEnum):
    """Scrape run outcome. Matches SQL: exit_status."""

    SUCCESS = "success"
    PARTIAL = "partial"
    ERROR = "error"
